firefox-ios/Client/Frontend/Home/LegacyHomepageViewController.swift

```diff
@@ -289,13 +289,20 @@ class LegacyHomepageViewController:
     }
 
     func createLayout() -> UICollectionViewLayout {
-        // swiftlint:disable line_length
-        let layout = UICollectionViewCompositionalLayout { [weak self] (sectionIndex: Int, layoutEnvironment: NSCollectionLayoutEnvironment) -> NSCollectionLayoutSection? in
-        // swiftlint:enable line_length
-            guard let self = self,
+        let zeroLayoutSize = NSCollectionLayoutSize(widthDimension: .absolute(0.0), heightDimension: .absolute(0.0))
+        let emptyGroup = NSCollectionLayoutGroup.horizontal(layoutSize: zeroLayoutSize,
+                                                            subitems: [NSCollectionLayoutItem(layoutSize: zeroLayoutSize)])
+        let emptyDefaultLayoutSection = NSCollectionLayoutSection(group: emptyGroup)
+
+        // swiftlint: disable line_length
+        let layout = UICollectionViewCompositionalLayout { [weak self] (sectionIndex: Int, layoutEnvironment: NSCollectionLayoutEnvironment)
+            -> NSCollectionLayoutSection? in
+        // swiftlint: enable line_length
+            guard let self,
                   let viewModel = self.viewModel.getSectionViewModel(shownSection: sectionIndex),
                   viewModel.shouldShow
-            else { return nil }
+            // Returning a default section to prevent the app to crash with invalid section definition
+            else { return emptyDefaultLayoutSection }
```

Comment:
Can you add a log with the logger here with a fatal level so we still see the issue show up in Sentry, it just won't crash on the user. This change can leave the user in a weird state, which is better than crashing but I'd still like to have some data on how often it's happening. Hopefully the home page rewrite can resolve the underlying issues causing this.

Reply:
Yes true, it make total sense so we can track the error on sentry 💪 . i'll log also the section that is causing that. I'd like to understand which section doesn't get loaded, by i believe could be only the ones who retrieve actual data like TopSites, JumpBackIn and bookmarks.